src/index.js

```diff
@@ -1,39 +1,124 @@
-    completed: false,
-  },
-  {
-    index: 2,
-    description: 'Challenging',
-    completed: true,
-  },
-];
-
-todoList.forEach((activity) => {
-  const { description, completed, index } = activity;
-
-  const cardList = document.querySelector('.card-list');
-
-  const li = document.createElement('li');
-  li.setAttribute('id', index);
-
-  const input = document.createElement('input');
-  input.setAttribute('type', 'checkbox');
-  input.checked = completed;
-  li.appendChild(input);
-
-  const p = document.createElement('p');
-  p.setAttribute('class', 'description');
-  p.textContent = description;
-  li.appendChild(p);
-
-  const i = document.createElement('i');
-  i.setAttribute('class', 'fa-solid fa-ellipsis-vertical');
-  li.appendChild(i);
-
-  cardList.appendChild(li);
-});
\ No newline at end of file
+class Tasks {
+  constructor(description, completed = false, index) {
+    this.description = description;
+    this.completed = completed;
+    this.index = index;
+  }
+}
+
+class Todolist {
+  constructor() {
+    this.taskDtata = [];
+  }
+
+  display = () => {
+    const taskitem = document.querySelector('.task-item');
+    if (taskitem) {
+      taskitem.innerHTML = '';
+    }
+    this.taskDtata.forEach((element) => {
+      const li = [];
+      li[element.index] = document.createElement('li');
+      li[element.index].setAttribute('id', element.index);
+      const p = [];
+      p[element.index] = document.createElement('p');
+      p[element.index].textContent = element.description;
+      p[element.index].setAttribute('id', element.index);
+      p[element.index].contentEditable = true;
+      const inputBox = [];
+      inputBox[element.index] = document.createElement('input');
+      inputBox[element.index].setAttribute('type', 'checkbox');
+      inputBox[element.index].classList.add('checkbox');
+      inputBox[element.index].setAttribute('id', element.index);
+      const button = [];
+      button[element.index] = document.createElement('button');
+      button[element.index].setAttribute('id', element.index);
+      button[element.index].innerHTML = '<i class="fa-solid fa-ellipsis-vertical icon"></i>';
+      li[element.index].append(inputBox[element.index], p[element.index], button[element.index]);
+      taskitem.append(li[element.index]);
+
+      p[element.index].addEventListener('click', (e) => {
+        e.target.nextSibling.innerHTML = '<i class="fa-solid fa-trash-can"></i>';
+        e.target.nextSibling.style.cursor = 'pointer';
+
+        e.target.nextSibling.addEventListener('click', () => {
+          li[element.index].remove();
+          this.removetask(element.index);
+        });
+      });
+
+      li[element.index].addEventListener('mouseleave', (e) => {
+        button[element.index].innerHTML = '<i class="fa-solid fa-ellipsis-vertical"></i>';
+        this.updatetask(e.target.id, e.target.innerText);
+      });
+    });
+  }
+
+  addtask = (description, completed, index) => {
```

Comment:
- [ ] As stated on the requirements, please add a new JS file for the new functionality 
![image](https://user-images.githubusercontent.com/62705947/220763445-21735fe5-6dc2-4e29-901f-da73b14eee47.png)